util/jsonrpc/jsonrpc.go

```diff
@@ -278,16 +275,16 @@ func (client *RPCClient) establishConnection(backend *Backend) error {
 	return nil
 }
 
-func (client *RPCClient) notify(status Status) {
-	for _, callback := range client.onConnectionStatusChangesNotify {
-		callback(status)
+func (client *RPCClient) notify(err error) {
+	for _, callback := range client.onConnectionErrorChangesNotify {
+		callback(err)
 	}
 }
 
-func (client *RPCClient) setStatus(status Status) {
-	if status != client.status {
-		client.status = status
-		go client.notify(status)
+func (client *RPCClient) setConnectionError(err error) {
+	if err != client.connectionError {
```

Comment:
Same here: a second `setConnectionError(errors.New("I'm offline"))` will result in an event notification but a `setConnectionError(io.EOF)` won't.